### PR TITLE
Use "exec" manifest for different action kinds

### PR DESCRIPTION
**Pull request by rabbah**
- Read exec manifest from deployment configuration to configure available runtimes
- Remove Java exec type.
- Normalize code as string vs attachment handling via a runtime manifest.

PG4/35.

### CODE REVIEW

File: tests/src/test/scala/whisk/core/entity/test/ExecHelpers.scala [hunk 1]
```diff
@@ -13,21 +13,37 @@
...
     protected val SWIFT = "swift"
     protected val SWIFT3 = "swift:3"
 
-    protected def js(code: String, main: Option[String] = None): Exec = CodeExecAsString(NODEJS, trim(code), main.map(_.trim), false)
-    protected def js6(code: String, main: Option[String] = None): Exec = CodeExecAsString(NODEJS6, trim(code), main.map(_.trim), false)
-    protected def swift(code: String, main: Option[String] = None): Exec = CodeExecAsString(SWIFT, trim(code), main.map(_.trim), true)
-    protected def swift3(code: String, main: Option[String] = None): Exec = CodeExecAsString(SWIFT3, trim(code), main.map(_.trim), false)
+    protected def js(code: String, main: Option[String] = None): Exec = CodeExecAsString(RuntimeManifest(NODEJS), trim(code), main.map(_.trim))
+    protected def js6(code: String, main: Option[String] = None): Exec = CodeExecAsString(RuntimeManifest(NODEJS6), trim(code), main.map(_.trim))
+    protected def swift(code: String, main: Option[String] = None): Exec = CodeExecAsString(RuntimeManifest(SWIFT, deprecated = Some(true)), trim(code), main.map(_.trim))
```
Comment (bjustin-ibm) [inline tests/src/test/scala/whisk/core/entity/test/ExecHelpers.scala]: If this refers to the Swift 2 runtime, then is it still needed here? It also appears in the [manifest deployment](https://github.com/openwhisk/openwhisk/blob/50aa4ee7a194b7b25d029cf827cb0ea00f5c1f2a/ansible/group_vars/all#L19), which seems confusing.

Comment (rabbah) [inline tests/src/test/scala/whisk/core/entity/test/ExecHelpers.scala]: It's deprecated; can delete or get. Can't invoke or update. These help to test it.

**Comment by rabbah** [comment]
PG2/1196 🔵

File: common/scala/src/main/scala/whisk/core/entity/Exec.scala [hunk 2]
```diff
@@ -174,60 +198,66 @@ protected[core] object Exec
...
+                    case _                => throw new DeserializationException(s"'image' must be a string defined in 'exec' for '${Exec.BLACKBOX}' actions")
+                }
+                val code: Option[String] = obj.getFields("code") match {
+                    case Seq(JsString(i)) => if (i.trim.nonEmpty) Some(i) else None
+                    case Seq(_)           => throw new DeserializationException(s"if defined, 'code' must a string defined in 'exec' for '${Exec.BLACKBOX}' actions")
+                    case _                => None
+                }
+                BlackBoxExec(image, code, optMainField)
+            } else {
+                throw new DeserializationException(s"kind '$kind' not in $runtimes")
```
Comment (markusthoemmes) [inline common/scala/src/main/scala/whisk/core/entity/Exec.scala]: I believe this is dead code. Rewrite the above to a `match` on kind would make this obvious I think.

File: common/scala/src/main/scala/whisk/core/entity/Exec.scala [hunk 3]
```diff
@@ -174,60 +198,66 @@ protected[core] object Exec
 
             val obj = v.asJsObject
 
-            val kindField = obj.getFields("kind") match {
+            val kind = obj.getFields("kind") match {
```
Comment (markusthoemmes) [inline common/scala/src/main/scala/whisk/core/entity/Exec.scala]: Use the `obj.fields.get("kind")` method like below to keep it consistent?

**Comment by rabbah** [comment]
addressed @markusthoemmes comments in https://github.com/openwhisk/openwhisk/pull/1974/commits/8d1f255e941cb5fc0cde52c8e2a7db752cfc083a. 

Rebasing to remove conflict.

**Comment by csantanapr** [comment]
merged 9a1398fb1a105b14fc86c26f9e406264ff25eea0

**Pull request closed without merging**